sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/v2ResolutionPlans.scala

```diff
@@ -244,3 +246,9 @@ case class ResolvedIdentifier(
     identifier: Identifier) extends LeafNodeWithoutStats {
   override def output: Seq[Attribute] = Nil
 }
+
+// A fake v2 catalog to hold temp views.
+object FakeSystemCatalog extends CatalogPlugin {
+  override def initialize(name: String, options: CaseInsensitiveStringMap): Unit = {}
+  override def name(): String = "SYSTEM"
```

Comment:
FAKE_SYSTEM?

Reply:
the name doesn't matter. We won't show it or look it up for now. But I think it's a good idea to add a system catalog officially, to host temp view, temp functions and builtin functions.